### PR TITLE
SAK-48009 Lessons Non-electronic Assignments should unlock

### PR DESCRIPTION
**Pull request by davidpbauer**
Adds submission type to the LessonsEntity to be checked when looking for item completion. Non-electronic assignments are "complete" immediately but still comply with the minimum grade rules.

### CODE REVIEW

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/AssignmentEntity.java [hunk 1]
```diff
@@ -87,6 +87,8 @@ public class AssignmentEntity implements LessonEntity, AssignmentInterface {
     public static final int CACHE_TIME_TO_LIVE_SECONDS = 600;
     public static final int CACHE_TIME_TO_IDLE_SECONDS = 360;
 
+    public static final int NON_ELECTRONIC_ASSIGNMENT_SUBMISSION = 4;
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/AssignmentEntity.java:90]: I'd use Assignment.SubmissionType.NON_ELECTRONIC_ASSIGNMENT_SUBMISSION.ordinal() here. I know it's more wordy, but I think this is the original and if it changes, you need to change this lessons specific one.

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonEntity.java [hunk 2]
```diff
@@ -119,6 +119,7 @@ public interface LessonEntity {
     public boolean isUsable();
     // only assignments
     public int getTypeOfGrade();
+    public int getSubmissionType();
```
Comment (adrianfish) [inline lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonEntity.java:122]: I'd add a default to this. It looks like you're returning 0 from the non implemented tools, so just do that here.
default public int getSubmissionType() {
    return 0;
}

You can then remove them from the non assignment lessons entities.

**Review by adrianfish — CHANGES_REQUESTED** [review]
Just a couple of things

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/AssignmentEntity.java (hunk 1 above)
Comment (davidpbauer) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/AssignmentEntity.java]: Good idea, thanks!

File: lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonEntity.java (hunk 2 above)
Comment (davidpbauer) [reply lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonEntity.java]: Didn't even occur to me to use default to cut down on changes needed. Thanks!

**Review by adrianfish — APPROVED** [review]
Nice, Dave. Thanks.

**Pull request merged**